controllers/nodefeaturediscovery_status.go

```diff
@@ -0,0 +1,567 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
```

Comment:
Hmm, I'm not sure we want to depend on downstream (openshift) at least this heavily 🧐   I think we should use k8s metav1 (`k8s.io/apimachinery/pkg/apis/meta/v1`).

Maybe `ConditionType`s is something that we could use from openshift 🤔  Or then just copy paste those here in order to break the dependency